### PR TITLE
Add items to order method.

### PR DESCRIPTION
**Pull request by phella**

### CODE REVIEW

File: services/order/service.go [hunk 1]
```diff
@@ -28,3 +51,153 @@ func (s orderService) CalculateUserCheck(context.Context) (int64, error) {
...
+		if err != nil {
+			return err
+		}
+
+		items[i] = models.OrderItem{
+			OrderID:    params.OrderID,
+			UserID:     params.UserID,
+			MealID:     meal.ID,
+			PriceE5:    priceE5,
+			Quantity:   meal.Quantity,
```
Comment (phella) [inline services/order/service.go:89]: use params instead of meal.

File: services/order/service.go [hunk 2]
```diff
@@ -28,3 +51,153 @@ func (s orderService) CalculateUserCheck(context.Context) (int64, error) {
...
+		newOrder := models.Order{
+			IsActive: pointers.Ptr(true),
+			TableId:  tableId,
+		}
+		res := s.db.Create(&newOrder)
+		if res.Error != nil {
+			return models.Order{}, res.Error
+		}
+		return newOrder, nil
+	}
```
Comment (OmarDesoky) [inline services/order/service.go]: How about using `s.db.FirstOrCreate`

File: services/order/service.go [hunk 3]
```diff
@@ -28,3 +51,153 @@ func (s orderService) CalculateUserCheck(context.Context) (int64, error) {
...
+			return 0, err
+		}
+		price += extraCharge
+	}
+	return price, nil
+}
+
+func findMealPriceE5ByID(meals []models.Meal, ID uint) (int64, error) {
+	fmt.Println(meals[0].ID)
+	fmt.Println(ID)
```
Comment (OmarDesoky) [inline services/order/service.go]: remove.

File: services/order/service.go [hunk 4]
```diff
@@ -28,3 +51,153 @@ func (s orderService) CalculateUserCheck(context.Context) (int64, error) {
...
+	for i, _ := range orderItems {
+		res[i] = OrderItem{
+			Meal:     orderItems[i].Meal,
+			User:     orderItems[i].User,
+			Quantity: orderItems[i].Quantity,
+		}
+	}
+
+	return res
+}
```
Comment (OmarDesoky) [inline services/order/service.go]: We can add a comment to move them to a new file/folder.

**Review by OmarDesoky — APPROVED** [review]
Great Work!

**Pull request merged**